File: 2330_valid_palindrome_iv/solution.cpp
// LeetCode 2330 - Valid Palindrome IV
// https://leetcode.com/problems/valid-palindrome-iv/

#include <string>

class Solution {
public:
    bool makePalindrome(std::string s) {
        int diff = 0;
        for (int i = 0, j = (int)s.size() - 1; i < j; ++i, --j) {
            if (s[i] != s[j]) {
                diff++;
                if (diff > 2) return false;
            }
        }
        return true;
    }
};

File: 2330_valid_palindrome_iv/solution.py
# LeetCode 2330 - Valid Palindrome IV
# https://leetcode.com/problems/valid-palindrome-iv/

class Solution:
    def makePalindrome(self, s: str) -> bool:
        diff = 0
        i, j = 0, len(s) - 1
        while i < j:
            if s[i] != s[j]:
                diff += 1
                if diff > 2:
                    return False
            i += 1
            j -= 1
        return True
